src/execution/executable_query.cpp

```diff
@@ -3,18 +3,26 @@
 #include "execution/ast/ast_dump.h"
 #include "execution/compiler/codegen.h"
 #include "execution/compiler/compiler.h"
+#include "execution/parsing/parser.h"
+#include "execution/parsing/scanner.h"
+#include "execution/sema/sema.h"
 #include "execution/util/region.h"
 #include "execution/vm/bytecode_generator.h"
 #include "execution/vm/module.h"
 #include "loggers/execution_logger.h"
 
 namespace terrier::execution {
 
+std::atomic<query_id_t> ExecutableQuery::query_identifier{query_id_t{0}};
```

Comment:
So this is just a global identifier? I'm not thrilled with globals but struggling to think of a better home for this. Can we at least declare it as `static` or do we need its symbol exported outside of this translation unit?

Reply:
I think this is already `static`, so we probably don't need this here. @17zhangw I think you can directly initialize it during declaration, right?

Reply:
`query_identifier` is declared as static within `ExecutableQuery`. I believe the C++ standard does not actually allow for initialization during declaration (or at least compilation fails).

https://stackoverflow.com/questions/9656941/why-cant-i-initialize-non-const-static-member-or-static-array-in-class

Reply:
Thanks, I missed the scoping on it.